test/pstl_offload/general/selected_different_device.pass.cpp

```diff
@@ -0,0 +1,43 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) 2023 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#if __SYCL_PSTL_OFFLOAD__ == 1
+#elif __SYCL_PSTL_OFFLOAD__ == 2
+    #undef __SYCL_PSTL_OFFLOAD__
+    #define __SYCL_PSTL_OFFLOAD__ 3
+#elif __SYCL_PSTL_OFFLOAD__ == 3
+    #undef __SYCL_PSTL_OFFLOAD__
+    #define __SYCL_PSTL_OFFLOAD__ 2
+#else
```

Comment:
Based on discussion above wow about this?

```suggestion
#if __SYCL_PSTL_OFFLOAD__ == 1
#elif __SYCL_PSTL_OFFLOAD__ == 2
    #undef __SYCL_PSTL_OFFLOAD__
    #define __SYCL_PSTL_OFFLOAD__ 3
#elif __SYCL_PSTL_OFFLOAD__ == 3
    #undef __SYCL_PSTL_OFFLOAD__
    #define __SYCL_PSTL_OFFLOAD__ 2
#elif __SYCL_PSTL_OFFLOAD != 1
```
Is it better than empty if or not? @kboyarinov, @Alexandr-Konovalov, opinions?

Reply:
I'm for `else` and for keeping all cases enumerated explicitly, as if someday we'll add yet another value for `__SYCL_PSTL_OFFLOAD__`, the test failed with clear diagnostics.

Reply:
Both is working for me, I'm abstaining.

Reply:
I would prefer the suggested option to avoid empty branches.
From the functional perspective, seems equivalent.

Reply:
Sure, let's change it.